influxql/iterator.go

```diff
@@ -86,6 +86,20 @@ func NewLimitIterator(input Iterator, opt IteratorOptions) Iterator {
 	}
 }
 
+// NewFillIterator returns an iterator that fills missing intervals with a value.
+func NewFillIterator(input Iterator, opt IteratorOptions) Iterator {
+	if opt.Fill == NoFill {
+		panic("iterator fill option must be set to something other than NoFill")
+	}
```

Comment:
Passing in with `NoFill` should just return `input` without wrapping it in a fill iterator.